app/validators/open_before_closed_validator.rb

```diff
@@ -0,0 +1,15 @@
+class OpenBeforeClosedValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, _value)
+    record.errors.add(attribute, error_message) unless before_closed?(record)
+  end
+
+private
+
+  def before_closed?(record)
+    record.opened_date <= record.closed_date || record.closed_date.blank?
+  end
```

Comment:
This method might raise an error if `record.opened_date` or `record.closed_date` are `nil` because the `<=` method isn't defined on `NilClass`. Would it be worth testing this case explicitly? We could potentially do those as unit tests for this validator if that's easier. What do you think?